File: Backend/models/Community.js
const mongoose = require('mongoose');

const communitySchema = new mongoose.Schema({
    name: { 
        type: String, 
        required: true,
        unique: true
    },
    description: { 
        type: String 
    },
    icon: { 
        type: String,
        default: '👥'
    },
    memberCount: {
        type: Number,
        default: 0
    },
    creator: {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'Account'
    },
    created_at: { 
        type: Date, 
        default: Date.now 
    }
});

module.exports = mongoose.model('Community', communitySchema);

File: App/forum/lib/models/community.dart
class Community {
  final String id;
  final String name;
  final String? description;
  final String icon;
  final int memberCount;
  final int postCount;

  Community({
    required this.id,
    required this.name,
    this.description,
    required this.icon,
    required this.memberCount,
    required this.postCount,
  });

  factory Community.fromJson(Map<String, dynamic> json) {
    return Community(
      id: json['_id'] ?? '',
      name: json['name'] ?? '',
      description: json['description'],
      icon: json['icon'] ?? '👥',
      memberCount: json['memberCount'] ?? 0,
      postCount: json['postCount'] ?? 0,
    );
  }
}

